src/databricks/labs/remorph/snow/databricks.py

```diff
@@ -87,7 +87,46 @@ def _select_contains_index(expression: exp.Select) -> bool:
     return False
 
 
+def _has_parse_json(expression):
+    if expression.find(exp.ParseJSON):
+        return True
+    _select = expression.find_ancestor(exp.Select)
+    if _select:
+        _from = _select.find(exp.From)
+        if _from:
+            _parse_json = _from.find(exp.ParseJSON)
+            if _parse_json:
+                return True
+    return False
+
+
+def _generate_function_str(select_contains_index, has_parse_json, generator_expr, alias, is_outer, alias_str):
+    if select_contains_index:
+        generator_function_str = f"POSEXPLODE({generator_expr})"
+        alias_str = f"{' ' + alias.name if isinstance(alias, exp.TableAlias) else ''} AS index, value"
+    elif has_parse_json and is_outer:
+        generator_function_str = f"VARIANT_EXPLODE_OUTER({generator_expr})"
+    elif has_parse_json:
+        generator_function_str = f"VARIANT_EXPLODE({generator_expr})"
+    else:
+        generator_function_str = f"VIEW EXPLODE({generator_expr})"
```

Comment:
@ganeshdogiparthi-db can you add couple of functional tests for Snowflake to cover view explode

Reply:
Added Sundar 
<img width="573" alt="Screenshot 2024-11-18 at 1 08 28 PM" src="https://github.com/user-attachments/assets/a94f2b2f-ab5e-439d-b5b6-f8dfe02cd523">
<img width="658" alt="Screenshot 2024-11-18 at 1 07 59 PM" src="https://github.com/user-attachments/assets/19666880-0657-4bf9-91b3-5d0a7dcfbbf5">
<img width="623" alt="Screenshot 2024-11-18 at 1 07 44 PM" src="https://github.com/user-attachments/assets/48b3f075-abb3-4de1-a01f-2ed97a121015">
<img width="699" alt="Screenshot 2024-11-18 at 1 07 38 PM" src="https://github.com/user-attachments/assets/bbe6b09a-ccc0-46a4-9557-f53faedb1576">
<img width="437" alt="Screenshot 2024-11-18 at 12 24 59 PM" src="https://github.com/user-attachments/assets/7e0df077-21ff-4654-a596-ea7a5c486d57">
<img width="726" alt="Screenshot 2024-11-18 at 12 24 53 PM" src="https://github.com/user-attachments/assets/fcf33eb8-2c62-484a-b98c-b48bfcfa9145">